### ISSUE TITLE
[Panel]: Label for Expand/Collapse Button in Panel component with Custom Header is not being announced properly by screen reader

### ISSUE THREAD

**Issue by OliwiaGowor** [SAP/ui5-webcomponents#10035]
### Bug Description

When using the Panel component with a custom header, the label read by the screen reader for the expand/collapse button is limited to 'Expand/Collapse, collapsed (/expanded), button.' There is no option to set or add additional information, such as the header title, for the screen reader to announce alongside this label.
![Screenshot 2024-10-15 at 14 58 49](https://github.com/user-attachments/assets/bb09f8b0-f6d1-41e2-b42a-092ae2e0ed3b)

### Affected Component

ui5-panel

### Expected Behaviour

On accessing the expand/collapse button in the Panel component with custom header, its respective label (the header content) must be read.

### Steps to Reproduce

1. Open UI5 Panel documentation
2. Open a Screen Reader
3. Navigate to Custom Header panel - the expand/collapse button
4. Screen reader doesn't read header title, only "Expand/Collapse, collapsed(/expanded), button"

### Priority

Medium

### UI5 Web Components Version

1.24.0, 2.3.0

### Browser

Chrome

### Operating System

MacOS

### Additional Context

Issue after SAP ATL team accessibility testing: https://github.com/kyma-project/busola/issues/3142

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by elenastoyanovaa** [comment]
Hello,

There is no technical way for us to "find" the title of the panel, when a custom header is provided by a slot. For this you need to use the `accessibleName` property of the ui5-panel. It adds a aria-label to the element which gets the `accessibleRole` of the ui5-panel - which is the component container. By doing so when focusing the panel, the speech output will provide the accessibleName passed as a property, the accessibleRole and the information regarding the focused element (in this case the expand/collapse button). If you insist that this label should be added to the button as well, you need to use `useAccessibleNameForToggleButton` private property. It is private because by setting the accessibleName to both the container and the button, this will lead to double announcement of the `accessibleName` on initial focusin in the ui5-panel.

You can read more about this here:
https://github.com/SAP/ui5-webcomponents/pull/2860

Kind Regards,
Elena
